File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        for (int i = 0; i < prices.length; i++) {
            int length = 0; 
            while ((i + length < prices.length) && (prices[i] <= prices[i + length])) {
                if (prices[i+length]-prices[i] > max) {
                    max = Math.max(max, prices[i+length]-prices[i]);
                }
                length++;
            }
        }
        System.out.println(max);
        return max;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n = len(nums)
        nset = set(nums)
        longest = 0
        for num in nums:
            if (num - 1) not in nset:
                length = 1
                while (num + length) in nset:
                    #this works bc count small to big
                    length +=1
                longest = max(length, longest)
        return longest
                
            



        
        